### PR TITLE
Resolved checkstyle and findbugs errors in Cannon

### PR DESCRIPTION
**Pull request by defnoob**

### CODE REVIEW

**Comment by defnoob** [comment]
Don't merge yet, still have to figure out how to make travis/maven fail on checkstyle violation.

**Comment by defnoob** [comment]
Now it's ready for merging.

File: bust-a-move-framework/src/main/java/bustamove/game/Cannon.java [hunk 1]
```diff
@@ -25,64 +26,38 @@
...
-    private static final int yLoad = (int) (530 - Bubble.DIAMETER / 2);
+    private static final int X_LOAD = (int) (320 - Bubble.DIAMETER / 2);
+    private static final int Y_LOAD = (int) (530 - Bubble.DIAMETER / 2);
     /**
      * The length of the cannon barrel.
      */
     private int length = GameConfig.DEFAULT_CANNON_LENGTH;
+
     /**
      * The color of the cannon barrel.
```
Comment (peace-4-you) [inline bust-a-move-framework/src/main/java/bustamove/game/Cannon.java]: Javadoc does not match with the variables underneath

File: bust-a-move-framework/src/main/java/bustamove/game/Cannon.java [hunk 2]
```diff
@@ -231,21 +210,23 @@ public final void update(final GameContainer container, final int delta) {
...
+     *
      * @param g graphics
      */
     public final void draw(final Graphics g) {
         g.setColor(this.cannonColour);
-        g.drawLine(Cannon.x, Cannon.y, (int) (Cannon.x + Math.cos(Math.toRadians(
-                this.angle + angleOffset)) * this.length), (int) (Cannon.y
-                - Math.sin(Math.toRadians(this.angle + angleOffset))
-                * this.length));
+        double rad = Math.toRadians(angle + ANGLE_OFFSET);
```
Comment (peace-4-you) [inline bust-a-move-framework/src/main/java/bustamove/game/Cannon.java]: Nice !

File: bust-a-move-framework/src/main/java/bustamove/player/Score.java [hunk 3]
```diff
@@ -1,11 +1,3 @@
-/*
- * File: Score.java
- * Class: Score
```
Comment (peace-4-you) [inline bust-a-move-framework/src/main/java/bustamove/player/Score.java]: Why is this removed ?

File: bust-a-move-framework/src/main/java/bustamove/App.java [hunk 4]
```diff
@@ -93,15 +93,6 @@ public static final Game getGame() {
     }
 
     /**
-     * Setter function for game object.
-     *
-     * @param g game object
-     */
```
Comment (peace-4-you) [inline bust-a-move-framework/src/main/java/bustamove/App.java]: just a question: removed because it is not used ?

File: bust-a-move-framework/src/main/java/bustamove/gamestate/StartScreen.java [hunk 5]
```diff
@@ -33,41 +34,46 @@ public final int getID() {
...
-     * @param game the game container
+     *
+     * @param game           the game container
      * @param stateBasedGame the state based game
      * @throws SlickException any type of slick exception
      */
     public final void init(final GameContainer game,
-        final StateBasedGame stateBasedGame) throws SlickException {
+                           final StateBasedGame stateBasedGame)
+            throws SlickException {
```
Comment (peace-4-you) [inline bust-a-move-framework/src/main/java/bustamove/gamestate/StartScreen.java]: Is it supposed to be empty ?

File: bust-a-move-framework/src/main/java/bustamove/bubble/Bubble.java [hunk 6]
```diff
@@ -378,6 +379,7 @@ public Circle getBoundingBox() {
      *
      * @param nextSPEED new speed value
      */
+    //CHECKSTYLE:OM: DesignForExtension
```
Comment (peace-4-you) [inline bust-a-move-framework/src/main/java/bustamove/bubble/Bubble.java]: Type ? OM -> ON

Comment (defnoob) [inline bust-a-move-framework/src/main/java/bustamove/bubble/Bubble.java]: Nice catch

File: bust-a-move-framework/src/main/java/bustamove/player/Score.java (hunk 3 above)
Comment (defnoob) [inline bust-a-move-framework/src/main/java/bustamove/player/Score.java]: I remember removing it because checkstyle marked it as red. Turns out the error wasn't there. I'll put it back

File: bust-a-move-framework/src/main/java/bustamove/App.java (hunk 4 above)
Comment (defnoob) [inline bust-a-move-framework/src/main/java/bustamove/App.java]: Wasn't used and was listed as a bug by FindBugs. We aren't supposed to write to a static field from an instance method. We also aren't supposed to instantiate the game object in there, but I left that since we are going to apply the singleton pattern on the game class anyways.

Comment (peace-4-you) [inline bust-a-move-framework/src/main/java/bustamove/App.java]: I see. Good

File: bust-a-move-framework/src/main/java/bustamove/gamestate/StartScreen.java (hunk 5 above)
Comment (peace-4-you) [inline bust-a-move-framework/src/main/java/bustamove/gamestate/StartScreen.java]: ok
